external/fv3fit/fv3fit/_shared/stacking.py

```diff
@@ -70,68 +34,6 @@ def stack_non_vertical(ds: xr.Dataset) -> xr.Dataset:
     return stack(ds=ds, unstacked_dims=Z_DIM_NAMES)
 
 
-def check_empty(ds: xr.Dataset) -> xr.Dataset:
-    """
-    Check for an empty variables along a dimension in a dataset
-    """
-    if len(ds[SAMPLE_DIM_NAME]) == 0:
-        raise ValueError("Check for NaN fields in the training data.")
-    return ds
-
-
-def preserve_samples_per_batch(ds: xr.Dataset) -> xr.Dataset:
```

Comment:
@spencerkclark will the loaders still do this for the multiple climate datasets when you specify subsampling and `unstacked_dims: z` for stacking during the batch loading?

Reply:
Thanks for checking -- in my latest runs I have stopped doing this, so I am good with removing it.